lib/board.rb

```diff
@@ -0,0 +1,51 @@
+class Board
+  attr_accessor :cells
+
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells.clear
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(index)
+    @cells[index.to_i - 1]
+  end
+
+  def full?
+    @cells.all? do |position|
+      position == "X" || position == "O"
+    end
+  end
```

Comment:
When using an iterator with only one line, convention is to use curly braces.  This is just style convention, and of course will not affect performance at all:

``` ruby
@cells.all? { |position| position == "X" || position == "O" }
```